lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -399,16 +391,32 @@ public function getReflectionService()
     }
 
     /**
-     * Gets the real class name of a class name that could be a proxy.
+     * Gets the real class name of a class name that could be a proxy or alias.
+     *
+     * @param string $className
+     *
+     * @return string
      */
-    private function getRealClass(string $class) : string
+    private function getRealClassName($className)
```

Comment:
Type hints are here to stay: there's no reason you should assume anything else than a string for `$className`.